### PR TITLE
replace a bunch of *mut i8 with rust strings

### PR DESCRIPTION
**Pull request by Mrmaxmeier**
~:construction: WIP :construction:~
This will probably fail CI. Let's see.

### CODE REVIEW

File: dpx/src/dpx_type1c.rs [hunk 1]
```diff
@@ -347,15 +329,7 @@ pub unsafe fn pdf_font_load_type1c(font: *mut pdf_font) -> i32 {
...
-    let fullname =
-        new((strlen(fontname).wrapping_add(8)).wrapping_mul(::std::mem::size_of::<i8>()) as _)
-            as *mut i8;
-    sprintf(
-        fullname,
-        b"%6s+%s\x00" as *const u8 as *const i8,
-        uniqueTag,
-        fontname,
-    );
+    let fullname = format!("{}+{}", uniqueTag, font.fontname);
```
Comment (burrbull) [inline dpx/src/dpx_type1c.rs:332]: "{:>6}+{}"

File: dpx/src/dpx_cmap.rs [hunk 2]
```diff
@@ -1006,70 +980,54 @@ unsafe fn check_range(
     }
     0i32
 }
-static mut __cache: *mut CMap_cache = std::ptr::null_mut();
+static mut __cache: Vec<Box<CMap>> = Vec::new();
```
Comment (burrbull) [inline dpx/src/dpx_cmap.rs:993]: Why not just `Vec<CMap>`?

File: dpx/src/dpx_cmap_write.rs [hunk 3]
```diff
@@ -328,20 +320,8 @@ pub unsafe fn CMap_create_stream(cmap: *mut CMap) -> Option<pdf_stream> {
...
-            ),
-        );
-        csi_dict.set(
-            "Ordering",
-            pdf_string::new_from_ptr(
-                (*csi).ordering as *const libc::c_void,
-                strlen((*csi).ordering) as _,
-            ),
-        );
+        csi_dict.set("Registry", (*csi).registry.into_obj());
```
Comment (burrbull) [inline dpx/src/dpx_cmap_write.rs]: `pdf_string` was here

File: dpx/src/dpx_cmap.rs (hunk 2 above)
Comment (Mrmaxmeier) [reply dpx/src/dpx_cmap.rs:993]: I'm currently boxing the cache entries to be able to get a stable `*mut` pointer even after Vec's backing storage is reallocated for resizing. These `Box`es should probably be replaced with `Rc`s soon.

File: dpx/src/dpx_type1c.rs (hunk 1 above)
Comment (Mrmaxmeier) [reply dpx/src/dpx_type1c.rs:332]: uniqueTag is always 6 characters long. `"{}+{}"` looks a bit cleaner.

**Comment by burrbull** [comment]
LGTM

**Comment by crlf0710** [comment]
There's a few `pub(crate)` that got changed into `pub` in this PR. Could you verify that these changes are necessary?(I believe not).

**Comment by Mrmaxmeier** [comment]
> There's a few `pub(crate)` that got changed into `pub` in this PR. Could you verify that these changes are necessary?(I believe not).

Your right. These changes are from a sloppy rebase. I believe I've got all of them in https://github.com/crlf0710/tectonic/commit/f7abe316829a51650f399a9a9707226daaf53a35.

**Pull request merged**